src/remote_adapter/remote_adapter_client/lua_lib/sdl_remote_adapter_client_lua_wrapper.cc

```diff
@@ -0,0 +1,383 @@
+#include "lua_lib/sdl_remote_adapter_client_lua_wrapper.h"
+
+#include <iostream>
+
+#include "sdl_remote_adapter_client.h"
+#include "rpc/detail/log.h"
+
+namespace lua_lib {
+
+RPCLIB_CREATE_LOG_CHANNEL(SDLRemoteClientLuaWrapper)
+
+int SDLRemoteClientLuaWrapper::create_SDLRemoteClient(lua_State* L) {
+  LOG_INFO("{0}",__func__);
+  luaL_checktype(L, 1, LUA_TTABLE);
+  // Index -1(top) - server port
+  // Index -2 - server ip
+  // index -3 - Library table
+
+  auto port = lua_tointeger(L, -1);
+  auto ip = lua_tostring(L, -2);
+  lua_pop(L, 2);  // Remove 2 values from top of the stack
+  // Index -1(top) - Library table
+
+  try {
+    SDLRemoteTestAdapterClient* client = new SDLRemoteTestAdapterClient(ip, port);
+
+    // Allocate memory for a pointer to client object
+    SDLRemoteTestAdapterClient** s =
+        (SDLRemoteTestAdapterClient**)lua_newuserdata(
+            L, sizeof(SDLRemoteTestAdapterClient*));
+    // Index -1(top) - instance userdata
+    // Index -2 - Library table
+
+    *s = client;
+  } catch (std::exception& e) {
+    std::cout << "Exception occurred: " << e.what() << std::endl;
+    lua_pushnil(L);
+    // Index -1(top) - nil
+    // Index -2 - Library table
+
+    return 1;
+  }
+
+  SDLRemoteClientLuaWrapper::registerSDLRemoteClient(L);
+  LOG_INFO("{0}",__func__);
+  // Index -1 (top) - registered SDLRemoteClient metatable
+  // Index -2 - instance userdata
+  // Index -3 - Library table
+
+  lua_setmetatable(L, -2);  // Set class table as metatable for instance userdata
+  // Index -1(top) - instance userdata
+  // Index -2 - Library table
+
+  return 1;
+}
+
+int SDLRemoteClientLuaWrapper::destroy_SDLRemoteClient(lua_State* L) {
+  LOG_INFO("{0}",__func__);
+  auto instance = get_instance(L);
+  delete instance;
+  return 0;
+}
+
+void SDLRemoteClientLuaWrapper::registerSDLRemoteClient(lua_State* L) {
+  LOG_INFO("{0}",__func__);
+  static const luaL_Reg SDLRemoteClientFunctions[] = {
+      {"connected", SDLRemoteClientLuaWrapper::lua_connected},  
+      {"app_start", SDLRemoteClientLuaWrapper::lua_app_start},
+      {"app_stop", SDLRemoteClientLuaWrapper::lua_app_stop},
+      {"app_check_status", SDLRemoteClientLuaWrapper::lua_app_check_status},
+      {"command_execute",SDLRemoteClientLuaWrapper::lua_command_execute},
+      {"file_exists", SDLRemoteClientLuaWrapper::lua_file_exists},
+      {"file_update", SDLRemoteClientLuaWrapper::lua_file_update},
+      {"file_content", SDLRemoteClientLuaWrapper::lua_file_content},
+      {"file_delete", SDLRemoteClientLuaWrapper::lua_file_delete},
+      {"file_backup", SDLRemoteClientLuaWrapper::lua_file_backup},
+      {"file_restore", SDLRemoteClientLuaWrapper::lua_file_restore},
+      {"folder_exists", SDLRemoteClientLuaWrapper::lua_folder_exists},
+      {"folder_create", SDLRemoteClientLuaWrapper::lua_folder_create},
+      {"folder_delete", SDLRemoteClientLuaWrapper::lua_folder_delete},
+      {"tcp_open", SDLRemoteClientLuaWrapper::lua_tcp_open},
```

Comment:
@SNiukalov What purpose is of providing "tcp_*" functions? 
Besides, as I can see `tcp_*` functions provide websocket communication functions instead. I propose to rename them accordingly.

Reply:
@aderiabin Fixed in 1826a1e